File: apps/go/internal/model/catalog.go
package model

import "encoding/json"

type CategoriesSummary struct {
	ID           ID      `db:"id" json:"id"`
	Name         string  `db:"name" json:"name"`
	Icon         *string `db:"icon" json:"icon,omitempty"`
	Img          *string `db:"img" json:"img,omitempty"`
	ProductCount int64   `db:"product_count" json:"product_count"`
} // @name Category

type BrandsSummary struct {
	ID           ID     `db:"id" json:"id"`
	Name         string `db:"name" json:"name"`
	ProductCount int64  `db:"product_count" json:"product_count"`
} // @name Brand

type ShippingMethod struct {
	ID      ID     `db:"id" json:"id"`
	Name    string `db:"name" json:"name"`
	CostIDR int64  `db:"cost_idr" json:"cost_idr"`
} // @name ShippingMethod

type PaymentMethod struct {
	ID       ID              `db:"id" json:"id"`
	Name     string          `db:"name" json:"name"`
	Metadata json.RawMessage `db:"metadata" json:"metadata"`
} // @name PaymentMethod

File: apps/js/src/model/catalog.ts
export type CategoriesSummary = {
	id: number;
	name: string;
	icon?: string;
	img?: string;
	product_count: number;
};

export type BrandsSummary = {
	id: number;
	name: string;
	product_count: number;
};

export type ShippingMethod = {
	id: number;
	name: string;
	cost_idr: number;
};

export type PaymentMethod = {
	id: number;
	name: string;
	metadata: Record<string, unknown>;
};
